op.c

```diff
@@ -9069,6 +9069,99 @@ Perl_newSTATEOP(pTHX_ I32 flags, char *label, OP *o)
     return op_prepend_elem(OP_LINESEQ, (OP*)cop, o);
 }
 
+/*
+=for apidoc cop_has_warning
+
+Returns true if the set of warnings bits contained by (or implied by) the
+COP contains the given warning, as specified by one of the C<WARN_...>
+constants from F<warnings.h>.
+
+=cut
+*/
+
+bool
+Perl_cop_has_warning(pTHX_ const COP *cop, int warn_bit)
+{
+    PERL_ARGS_ASSERT_COP_HAS_WARNING;
+
+    char *warning_bits = cop->cop_warnings;
+    if(warning_bits == pWARN_ALL)
+        return true;
+    else if(warning_bits == pWARN_NONE)
+        return false;
+    else
+        return isWARN_on(cop->cop_warnings, warn_bit);
+}
+
+#define cop_inplace_expand_warning_bitmask(cop)  S_cop_inplace_expand_warning_bitmask(aTHX_ cop)
+STATIC void
+S_cop_inplace_expand_warning_bitmask(pTHX_ COP *cop)
+{
+    char *warning_bits = cop->cop_warnings;
+
+    if(warning_bits == pWARN_STD || warning_bits == pWARN_ALL)
+        warning_bits = WARN_ALLstring;
```

Comment:
This looks wrong - shouldn't it distinguish STD from ALL?

Though I think you'd need to modify `regen/warnings.pl` to add `WARN_STDstring` to `warnings.h`